python/pipeline/stack.py

```diff
@@ -1029,16 +1016,19 @@ def surface_eqn(data, a, b, c, d, f):
         upper_threshold_percent = 0.6  # Surface median intensity should be in the bottom X% of the *range* of medians
         gaussian_blur_size = 5  # Size of gaussian blur applied to slice
         min_points_allowed = 10  # If there are less than X points after filtering, throw an error
-        bounds = (
-        [0, 0, np.NINF, np.NINF, np.NINF], [np.Inf, np.Inf, np.Inf, np.Inf, np.Inf])  # Bounds for paraboloid fit
+        bounds = ([0, 0, np.NINF, np.NINF, np.NINF], [np.Inf, np.Inf, np.Inf, np.Inf, np.Inf])  # Bounds for paraboloid fit
         ss_percent = 0.40  # Percentage of points to subsample for robustness check
         num_iterations = 1000  # Number of iterations to use for robustness check
 
+        if 'surface_method_id' not in key:
```

Comment:
no need to check for this, surface_method_id will always be there

Reply:
If I run populate and pass a stack key, how will the surface method be automatically filled out? Perhaps I'm missing something in my code.

Reply:
if you call Surface.populate(my_restr), the key passed to the make will be all in (PreprocessedStack * shared.SurfaceMethod & key).proj().fetch()